Kernel/Bus/PCI/Device.cpp

```diff
@@ -44,13 +44,23 @@ void Device::disable_message_signalled_interrupts()
 {
     TODO();
 }
+
 void Device::enable_extended_message_signalled_interrupts()
 {
-    TODO();
+    for (auto& capability : m_pci_identifier->capabilities()) {
+        if (capability.id().value() == PCI::Capabilities::ID::MSIX) {
+            capability.write16(2, capability.read16(2) | msix_control_enable);
```

Comment:
Is the offset of 2 here a magic number, what relevance does it have? Should probably be defined centrally somewhere.